src/components/recurring-donation/Form.tsx

```diff
@@ -0,0 +1,161 @@
+import React from 'react'
+import { useMutation, useQueryClient, UseQueryResult } from 'react-query'
+import { useRouter } from 'next/router'
+import { useTranslation } from 'next-i18next'
+import Link from 'next/link'
+import { AxiosError, AxiosResponse } from 'axios'
+import * as yup from 'yup'
+import { Box, Button, Grid, Typography } from '@mui/material'
+
+import { RecurringDonationInput, RecurringDonationResponse } from 'gql/recurring-donation'
+
+import { Currency } from 'gql/currency'
+import { useRecurringDonation } from 'common/hooks/recurringDonation'
+import { routes } from 'common/routes'
+import { ApiErrors } from 'service/apiErrors'
+import { useCreateRecurringDonation, useEditRecurringDonation } from 'service/recurringDonation'
+import { endpoints } from 'service/apiEndpoints'
+import { AlertStore } from 'stores/AlertStore'
+import GenericForm from 'components/common/form/GenericForm'
+import FormTextField from 'components/common/form/FormTextField'
+import SubmitButton from 'components/common/form/SubmitButton'
+import CurrencySelect from 'components/currency/CurrencySelect'
+import RecurringDonationStatusSelect from './grid/RecurringDonationStatusSelect'
+
+export enum RecurringDonationStatus {
+  trialing = 'trialing',
+  active = 'active',
+  canceled = 'canceled',
+  incomplete = 'incomplete',
+  incompleteExpired = 'incompleteExpired',
+  pastDue = 'pastDue',
+  unpaid = 'unpaid',
+}
+
+const validCurrencies = Object.keys(Currency)
+const validStatuses = Object.keys(RecurringDonationStatus)
+
+const validationSchema = yup
+  .object()
+  .defined()
+  .shape({
+    status: yup.string().oneOf(validStatuses).required(),
+    personId: yup.string().trim().max(50).required(),
+    extSubscriptionId: yup.string().trim().max(50).required(),
+    extCustomerId: yup.string().trim().max(50).required(),
+    amount: yup.number().positive().integer().required(),
+    currency: yup.string().oneOf(validCurrencies).required(),
+    sourceVault: yup.string().trim().uuid().required(),
+  })
+
+export default function EditForm() {
+  const router = useRouter()
+  const queryClient = useQueryClient()
+  const { t } = useTranslation()
+
+  let id = router.query.id
+
+  let initialValues: RecurringDonationInput = {
+    status: '',
+    personId: '',
+    extSubscriptionId: '',
+    extCustomerId: '',
+    amount: 0,
+    currency: '',
+    sourceVault: '',
+  }
+
+  if (id) {
+    id = String(id)
+    const { data }: UseQueryResult<RecurringDonationResponse> = useRecurringDonation(id)
+
+    initialValues = {
+      status: data?.status,
+      personId: data?.personId,
+      extSubscriptionId: data?.extSubscriptionId,
+      extCustomerId: data?.extCustomerId,
+      amount: data?.amount,
+      currency: data?.currency,
+      sourceVault: data?.sourceVault,
+    }
+  }
```

Comment:
These values should be pass from the page

The idea is:

1. In page load the initial recoerds
2. Load the form with the initial records predefined
3. Work on the data

At the moment `CreatePage` and `EditPage` look the same, but they should not.

See as example 

https://github.com/podkrepi-bg/frontend/blob/621cb6bb8d1ab4f658effb08ee9e300b8dfc35e8/src/components/campaigns/grid/EditPage.tsx